ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-commands.sh

```diff
@@ -153,37 +140,73 @@ ibm_cloud_cis_crn = "${IBMCLOUD_CIS_CRN}"
+  curl -L -o "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform.gz -L https://releases.hashicorp.com/terraform/"${TERRAFORM_VERSION}"/terraform_"${TERRAFORM_VERSION}"_linux_amd64.zip \
+    && gunzip "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform.gz \
+    && chmod +x "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform \
+    || true
+
+  if [ ! -f "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform ]
+  then
+    echo "Manually building the terraform code"
+    # upgrade go to GO_VERSION
+    if [ -z "$(command -v go)" ]
+    then
+      echo "go is not installed, proceed to installing go"
+      cd "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/ && wget -q https://go.dev/dl/go"${GO_VERSION}".linux-amd64.tar.gz && tar -C /tmp -xzf go"${GO_VERSION}".linux-amd64.tar.gz \
+        && export PATH=$PATH:/tmp/go/bin && export GOCACHE=/tmp/go && export GOPATH=/tmp/go && export GOMODCACHE="/tmp/go/pkg/mod"
+      if [ -z "$(command -v go)" ]
+      then
+        echo "Installed go successfully"
+      fi
+    fi
+
+    # build terraform from source using TERRAFORM_VERSION
+    echo "terraform is not installed, proceed to installing terraform"
+    cd "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/ && curl -L https://github.com/hashicorp/terraform/archive/refs/tags/v"${TERRAFORM_VERSION}".tar.gz -o "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform.tar.gz \
+      && tar -xzf "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform.tar.gz && cd "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform-"${TERRAFORM_VERSION}" \
+      && go build -ldflags "-w -s -X 'github.com/hashicorp/terraform/version.dev=no'" -o bin/ . && cp bin/terraform "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform
+  fi
+  echo "Applying terraform"
+  "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform apply -auto-approve -var-file var-multi-arch-upi.tfvars | \
+      sed '/.*client-certificate-data*/d; /.*token*/d; /.*client-key-data*/d; /- name: /d; /Login to the console with user/d' | \
+        while read LINE
+        do
+            if [[ "${LINE}" == *"BEGIN RSA PRIVATE KEY"* ]]
+            then
+            OUTPUT=""
+            fi
+            if [ ! -z "${OUTPUT}" ]
+            then
+                echo "${LINE}"
+            fi
+            if [[ "${LINE}" == *"END RSA PRIVATE KEY"* ]]
+            then
+            OUTPUT="yes"
+            fi
+        done
+  echo "POWERVS CLUSTER IS COMPLETE, log is redacted"
+  echo "Find ocp4-upi-powervs from root folder"
+  find /tmp/ -iname ocp4-upi-powervs
+  echo "Find all tfvars from root folder"
+  find /tmp/ -iname '*.tfvars'
+  cp "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform.tfstate "${SHARED_DIR}"/terraform-multi-arch-upi.tfstate
+  "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform output -raw -no-color bastion_private_ip | tr -d '"' > "${SHARED_DIR}"/BASTION_PRIVATE_IP
+  "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform output -raw -no-color bastion_public_ip | tr -d '"' > "${SHARED_DIR}"/BASTION_PUBLIC_IP
```

Comment:
```suggestion
  "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform output -raw -no-color bastion_private_ip | tr -d '"' > "${SHARED_DIR}"/BASTION_PRIVATE_IP
  "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/terraform output -raw -no-color bastion_public_ip | tr -d '"' > "${SHARED_DIR}"/BASTION_PUBLIC_IP
```

make sure you have the right directory context.